src/Core/src/Platform/Windows/ContentPanel.cs

```diff
@@ -104,27 +118,38 @@ void UpdateContent()
 			Content.RenderTransform = new TranslateTransform() { X = -strokeThickness, Y = -strokeThickness };
 		}
 
-		void UpdateClip(IShape? borderShape)
+		void UpdateClip(IShape? borderShape, double width, double height)
 		{
 			if (Content == null)
 				return;
+			
+			if (height <= 0 && width <= 0)
+				return;
 
 			var clipGeometry = borderShape;
 
 			if (clipGeometry == null)
 				return;
 
-			double width = ActualWidth;
-			double height = ActualHeight;
-
-			if (height <= 0 && width <= 0)
-				return;
-
 			var visual = ElementCompositionPreview.GetElementVisual(Content);
 			var compositor = visual.Compositor;
 
 			var pathSize = new Graphics.Rect(0, 0, width, height);
-			var clipPath = clipGeometry.PathForBounds(pathSize);
+
+			PathF? clipPath;
+
+			if (clipGeometry is IRoundRectangle roundedRectangle)
```

Comment:
Why just the round rects? What makes this special? Will we soon see other types being added?

Reply:
No idea, not my design. I was just making it work the way it worked on the other two platforms.